React/Fabric/Mounting/ComponentViews/ScrollView/RCTScrollViewComponentView.mm

```diff
@@ -679,6 +707,81 @@ - (void)removeScrollListener:(NSObject<UIScrollViewDelegate> *)scrollListener
   [self.scrollViewDelegateSplitter removeDelegate:scrollListener];
 }
 
+#pragma mark - Maintain visible content position
+
+- (void)_prepareForMaintainVisibleScrollPosition
+{
+  const auto &props = *std::static_pointer_cast<const ScrollViewProps>(_props);
+  if (!props.maintainVisibleContentPosition) {
+    return;
+  }
+
+  BOOL inverted = false; // TODO: props.inverted
+  BOOL horizontal = _scrollView.contentSize.width > self.frame.size.width;
+  int minIdx = props.maintainVisibleContentPosition.value().minIndexForVisible;
+  for (NSUInteger ii = minIdx; ii < _contentView.subviews.count; ++ii) {
+    // Find the first entirely visible view. This must be done after we update the content offset
```

Comment:
Conceptually, shouldn't this be finding the first entirely visible view _before_ we update the content offset?